### ISSUE TITLE
Map Tiles Question

### ISSUE THREAD

**Issue by kaldri** [luferau/labview-offline-maps#1]
Hello! Where did you download your map tiles from? When I zoom into the map, the screen goes blank (I'm assuming this is because the provided tiles don't have enough detail to zoom in that far).  Thanks!

**Comment by luferau** [comment]
Hey kaldri,
Map tiles you can get from software like http://www.sasgis.org/download/
Last realize: http://www.sasgis.org/programs/sasplanet/SASPlanet_201212.zip
Maps settings (should be copied to program maps folder): http://parasite.kicks-ass.org:3000/sasgis/maps/archive/master.zip

Map tiles would be saved to <program path>/cache/*** folders for different maps providers.

**Comment by jkreienh** [comment]
Hey @luferau, 

Nice work on the app! Can you provide the steps and settings to export the tiles from SASPlanet in png file format in the folder structure that you provided that is consumable by this app?  I see a single folder containing png files with x_y_z.png and I'm not sure how to get the export in this format. - Thanks

![image](https://user-images.githubusercontent.com/34135562/152654092-5fa95caf-2c33-435b-ac2d-4b8716f0958d.png)

**Comment by luferau** [comment]
Hello @jkreienh 

Visual Studio solution **vs\OfflineMapsSolution.sln** contain windows console application project **OfflineMaps.Console**

This application has **convert** command and can be run like:

`OfflineMaps.Console.exe convert -s g:\SAS.Planet.Release.190707\cache\yasat\ -d g:\OfflineMaps`

This command just copy files from SAS.Planet folder structure to one folder with specific file names. See following.

Map tiles format description
Source format: 
`z{zoom}\{x_div_1024}\x{x}\{y_div_1024}\y{y}`

Destination format:  
X {0} and Y {1} numbers and the Zoom level {2}
`{0}_{1}_{2}`

Ping me if you don't have Visual Studio and you need OfflineMaps.Console.exe binary files

**Comment by IvanLisRus** [comment]
Hey!
Can you send the file OfflineMaps.Console.exe
Or lay it out as part of the project.

**Comment by luferau** [comment]
@IvanLisRus, please see [Releases page ](https://github.com/luferau/labview-offline-maps/releases)

**Comment by IvanLisRus** [comment]
Спасибо!

**Comment by slimebdx** [comment]
> Hello @jkreienh
> 
> Visual Studio solution **vs\OfflineMapsSolution.sln** contain windows console application project **OfflineMaps.Console**
> 
> This application has **convert** command and can be run like:
> 
> `OfflineMaps.Console.exe convert -s g:\SAS.Planet.Release.190707\cache\yasat\ -d g:\OfflineMaps`
> 
> This command just copy files from SAS.Planet folder structure to one folder with specific file names. See following.
> 
> Map tiles format description Source format: `z{zoom}\{x_div_1024}\x{x}\{y_div_1024}\y{y}`
> 
> Destination format: X {0} and Y {1} numbers and the Zoom level {2} `{0}_{1}_{2}`
> 
> Ping me if you don't have Visual Studio and you need OfflineMaps.Console.exe binary files

thank you @luferau  it's work !! but only with the old version of sas planet like SAS.Planet.190725 for example , because if you use the newest version it's not work

**Comment by slimebdx** [comment]
@luferau please tell me how i can change the point information content (speed, altitude ) i need to change it to other informations like wind_speed ..ect
i need to make change in the dll or just labview

**Comment by luferau** [comment]
> @luferau please tell me how i can change the point information content (speed, altitude ) i need to change it to other informations like wind_speed ..ect i need to make change in the dll or just labview

@slimebdx what you are talking about is .NET dll responsibility. This is just an example and parameters list is hardcoded. _OfflineMapControl.AddPoint_ method in _OfflineMaps.Controls.OfflineMapControl.cs_ can be changed for using different parameters or another method can be created where you can pass arbitrary parameters.

**Comment by ajayavgarde** [comment]
hey, should I select the copy or export option in SAS.planet application?

**Comment by luferau** [comment]
No, map tiles are saved to SAS directory/**cache**/*** folders for different maps providers.
